packages/devtools_app/lib/src/service/vm_service_wrapper.dart

```diff
@@ -427,15 +427,15 @@ class VmServiceWrapper implements VmService {
   Future<SourceReport> getSourceReport(
     String isolateId,
     List<String> reports, {
-    String scriptId,
-    int tokenPos,
-    int endTokenPos,
-    bool forceCompile,
-    bool reportLines,
+    String? scriptId,
+    int? tokenPos,
+    int? endTokenPos,
+    bool? forceCompile,
+    bool? reportLines,
   }) async {
     // Workaround for https://github.com/flutter/devtools/issues/2981.
     // TODO(bkonyi): remove after Flutter stable is on Dart SDK > 2.12.
-    if (await serviceManager.connectedApp.isProfileBuild) {
+    if (await serviceManager.connectedApp!.isProfileBuild) {
```

Comment:
based on the above comment, I think you can remove this whole if block

Reply:
done